openflights.js

```diff
@@ -2234,215 +2246,223 @@ function exportFlights(type, newWindow) {
-    table +=
-      "<tr><th colspan=2>" + gt.gettext("Journey records") + "</th></tr>";
-    // TODO: Shouldn't be needed when stats.php returns full JSON
-    // https://github.com/jpatokal/openflights/issues/1296
-    var rows = longshort !== "" ? longshort.split(";") : [];
-    for (var r = 0; r < rows.length; r++) {
-      var col = rows[r].split(",");
-      // desc 0, distance 1, duration 2, s.iata 3, s.apid 4, d.iata 5, d.apid 6
-      table +=
-        "<tr><td>" +
-        col[0] +
-        '</td><td><a href="#" onclick="JavaScript:selectAirport(' +
-        col[4] +
-        ');">' +
-        col[3] +
-        '</a>&harr;<a href="#" onclick="JavaScript:selectAirport(' +
-        col[6] +
-        ');">' +
-        col[5] +
-        "</a>, " +
-        // TODO: Use more specific locale if one exists
-        col[1].toLocaleString("en-US") +
-        ", " +
-        col[2] +
-        "</td></tr>";
-    }
-    table +=
-      "<tr><td>" +
-      gt.gettext("Average") +
-      "</td><td>" +
-      // TODO: Use more specific locale if one exists
-      uniques["avg_distance"].toLocaleString("en-US") +
-      ", " +
-      uniques["avg_duration"] +
-      "</td></tr>";
-    table += "<tr><td>&nbsp;</td></tr>";
-    table += "<tr><td>&nbsp;</td></tr>";
-    table +=
-      "<tr><th colspan=2>" + gt.gettext("Airport records") + "</th></tr>";
-    // TODO: Shouldn't be needed when stats.php returns full JSON
-    // https://github.com/jpatokal/openflights/issues/1296
-    var rows = extremes !== "" ? extremes.split(":") : [];
-    for (r = 0; r < rows.length; r++) {
-      var col = rows[r].split(",");
-      // 0 desc, 1 code, 2 apid, 3 x, 4 y
-      var lat = parseFloat(col[4]).toFixed(2);
-      var lon = parseFloat(col[3]).toFixed(2);
-      if (lat < 0) {
-        lat = -lat + "&deg;S";
-      } else {
-        lat += "&deg;N";
-      }
-      if (lon < 0) {
-        lon = -lon + "&deg;W";
-      } else {
-        lon += "&deg;E";
-      }
-      table +=
-        "<tr><td>" +
-        col[0] +
-        '</td><td><a href="#" onclick="JavaScript:selectAirport(' +
-        col[2] +
-        ');">' +
-        col[1] +
-        "</a> (" +
-        lat +
-        " " +
-        lon +
-        ")</td></tr>";
-    }
-    table += "</table>";
-    bigtable += table + '</td><td style="vertical-align: top">';
+  if (statsData.total.segments === 0) {
+    result.append(gt.gettext("This user has no flights."));
+    return;
+  }
 
-    table = '<table style="border-spacing: 10px 0px">';
-    table +=
-      "<tr><th>" +
-      gt.gettext("Class") +
-      "</th><th>" +
-      gt.gettext("Reason") +
-      "</th><th>" +
-      gt.gettext("Seats") +
-      "</th></tr>";
-    table += "<tr>";
-    table += '<td><div id="chart_class"></div></td>';
-    table += '<td><div id="chart_reason"></div></td>';
-    table += '<td><div id="chart_seat"></div></td>';
-    table += "</tr>";
+  const displayUnit = gt.gettext(distance_units[statsData.distance_unit]);
```

Comment:
The `gt.gettext()` call shouldn't be needed as the localised unit message is already in `distance_units[statsData.distance_unit]`?

Reply:
Removed.